src/components/Company/PaySchedule/_parts/Edit.tsx

```diff
@@ -0,0 +1,120 @@
+import { useFormContext } from 'react-hook-form'
+import { ListBoxItem } from 'react-aria-components'
+import { useTranslation } from 'react-i18next'
+import { useEffect } from 'react'
+import { PayScheduleInputs, usePaySchedule } from '../PaySchedule'
+import { PayPreviewCard } from './PayPreviewCard/PayPreviewCard'
+import style from './Edit.module.scss'
+import Spinner from '@/assets/icons/spinner_small.svg?react'
+import { DatePicker } from '@/components/Common/Inputs/DatePicker'
+import { Flex, TextField, Select, RadioGroup, NumberField, Grid } from '@/components/Common'
+
+export const Edit = () => {
+  const { t } = useTranslation('Company.PaySchedule')
+  const { payPeriodPreview, mode, payPreviewLoading } = usePaySchedule()
+  const { control, watch, setValue } = useFormContext<PayScheduleInputs>()
+
+  const frequency = watch('frequency')
+  const customTwicePerMonth = watch('custom_twice_per_month')
+
+  const shouldShowDay1 =
+    (frequency === 'Twice per month' && customTwicePerMonth === 'true') || frequency === 'Monthly'
+  const shouldShowDay2 = frequency === 'Twice per month' && customTwicePerMonth === 'true'
+
+  useEffect(() => {
+    if (frequency === 'Twice per month' && customTwicePerMonth === 'false') {
+      setValue('day_1', 15)
+      setValue('day_2', 31)
+    }
+  }, [frequency, customTwicePerMonth, setValue])
+
+  if (mode !== 'EDIT_PAY_SCHEDULE' && mode !== 'ADD_PAY_SCHEDULE') {
+    return null
+  }
+
+  return (
+    <div className={style.payScheduleContainer}>
+      <Grid gap={4} gridTemplateColumns={{ base: '1fr', small: '1fr 1fr' }}>
+        <div className={style.payScheduleForm}>
+          <Flex flexDirection={'column'}>
+            <TextField control={control} name="custom_name" label="Name" />
+            <Select
+              control={control}
+              name="frequency"
+              label={t('labels.frequency')}
+              items={[
+                { id: 'Every week', name: t('frequencies.everyWeek') },
+                { id: 'Every other week', name: t('frequencies.everyOtherWeek') },
+                { id: 'Twice per month', name: t('frequencies.twicePerMonth') },
+                { id: 'Monthly', name: t('frequencies.monthly') },
+              ]}
+            >
+              {option => <ListBoxItem>{option.name}</ListBoxItem>}
+            </Select>
+            {frequency === 'Twice per month' && (
+              <RadioGroup
+                control={control}
+                name="custom_twice_per_month"
+                label={t('labels.frequencyOptions')}
+                description={t('descriptions.frequencyOptionsDescription')}
+                options={[
+                  { value: 'false', label: t('frequencyOptions.15thAndLast') },
+                  { value: 'true', label: t('frequencyOptions.custom') },
```

Comment:
Naming nit: the string true and false gave me pause for a sec if there was a typo 😅

Reply:
I changed them to something else since RHF isn't a big fan of booleans either so I swapped them for more specific strings.